libraries/chain/db_update.cpp

```diff
@@ -390,7 +390,7 @@ void database::clear_expired_orders()
          price settlement_price = pays / receives;
 
          // Calculate fill_price with a bigger volume to reduce impacts of rounding
-         if( settlement_fill_price.base.asset_id != pays.asset_id )
+         if( settlement_fill_price.base.asset_id != current_asset ) // only calculate once per asset
```

Comment:
I think this change is actually wrong. @oxarbitrage: try revert this and test again.

Reply:
Or change it to `if( settlement_fill_price.quote.asset_id != current_asset )`.

Reply:
Sorry, should have no issue here.